### PR TITLE
refactor(ui): restyle the transaction details modal

### PR DESCRIPTION
**Pull request by brandonhaslegs**
# Still to do:
- [x] Add timestamp
- [x] make timestamp display actual time
image
---
- [x] Add icon section to the top in the grey header section
- [x] make the icon based on the type of transaction
image
---
- [x] Color the RAD symbol red
- [x] reposition RAD symbol
- [x] show USD on hover
- [x] color deposit amount grey instead of red
image
---
- [x] Make transaction ID copyable
image
---
- [x] Reformat the transaction status to be like new design (this one was particularly tricky, I couldn't figure out how to move it without breaking something)
image

### CODE REVIEW

**Comment by rudolfs** [comment]
The transaction summary view now shows an empty transaction ID field:
Screenshot 2020-07-07 at 11 01 29.

Also the tests have to be fixed before moving this to review. Let me know if you need help with that @brandonhaslegs.

File: ui/Screen/Org/MemberRegistration.svelte [hunk 1]
```diff
@@ -69,35 +69,37 @@
...
-        transactionDeposits={$session.data.transactionDeposits} />
-    </div>
-  {/if}
-  <NavigationButtons
-    style="margin-top: 32px;"
-    {disableSubmit}
-    on:submit={next}
-    on:cancel={cancel} />
-</StepModalLayout>
+<ModalLayout dataCy="page">
```
Comment (juliendonck) [inline ui/Screen/Org/MemberRegistration.svelte:72]:
```suggestion
<ModalLayout dataCy="add-member-modal">
```

**Comment by juliendonck** [comment]
@rudolfs I fixed the tests myself!!

File: ui/DesignSystem/Component/Transaction/Header.svelte [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      case SubjectType.User:
+      case SubjectType.Member:
+      case SubjectType.UserProject:
+        return "circle";
+      default:
+        return "square";
+    }
+  };
+
+  const icon = () => {
```
Comment (rudolfs) [inline ui/DesignSystem/Component/Transaction/Header.svelte:26]: You don't need this function at all:

```diff
diff --git a/ui/DesignSystem/Component/Transaction/Header.svelte b/ui/DesignSystem/Component/Transaction/Header.svelte
index 6e78329..60aaf3b 100644
--- a/ui/DesignSystem/Component/Transaction/Header.svelte
+++ b/ui/DesignSystem/Component/Transaction/Header.svelte
@@ -22,20 +22,6 @@
         return "square";
     }
   };
-
-  const icon = () => {
-    const val = headerIcon(transaction.messages[0]);
-    switch (val) {
-      case "Project":
-        return Icon.Source;
-      case "User":
-        return Icon.Member;
-      case "Register":
-        return Icon.Register;
-      default:
-        return Icon.Register;
-    }
-  };
 </script>

 <style>
@@ -61,7 +47,7 @@
 <div data-cy="summary" variant={transaction.id ? 'top' : 'single'}>
   <div data-cy="message" class="container">
     <div class="icon">
-      <svelte:component this={icon()} style="fill: var(--color-primary)" />
+      <svelte:component this={Icon[headerIcon(transaction.messages[0])]} style="fill: var(--color-primary)" />
     </div>
     <Title variant="big">{formatMessage(transaction.messages[0])}</Title>
     <caption style="display: inline-flex; margin:16px 0;">
diff --git a/ui/src/transaction.ts b/ui/src/transaction.ts
index 0f2d076..552dd71 100644
--- a/ui/src/transaction.ts
+++ b/ui/src/transaction.ts
@@ -277,22 +277,16 @@ export const formatMessage = (msg: Message): string => {
 export const headerIcon = (msg: Message): string => {
   switch (msg.type) {
     case MessageType.OrgRegistration:
-      return "Register";
-
     case MessageType.OrgUnregistration:
       return "Register";

-    case MessageType.MemberRegistration:
-      return "User";
-
     case MessageType.MemberUnregistration:
-      return "User";
+    case MessageType.MemberRegistration:
+    case MessageType.UserRegistration:
+      return "Member";


     case MessageType.ProjectRegistration:
-      return "Project";
-
-    case MessageType.UserRegistration:
-      return "User";
+      return "Source";
   }
 };
```

File: ui/DesignSystem/Primitive/Title.svelte [hunk 3]
```diff
@@ -2,7 +2,7 @@
   export let style = null;
   export let dataCy = null;
 
-  export let variant = "regular"; // regular | medium | large | big | huge
+  export let variant = "regular"; // regular | tiny | medium | large | big | huge
```
[the remark this answers is not in the document]
Comment (brandonhaslegs) [reply ui/DesignSystem/Primitive/Title.svelte:5]: large = big and medium = regular typically. The naming is really weird. But tiny is definitely smaller than regular.

[the remark this answers is not in the document]
Comment (rudolfs) [reply ui/DesignSystem/Primitive/Title.svelte:5]: @brandonhaslegs could you come up with a good naming scheme for sizing? Would love to improve that in our codebase.

File: ui/DesignSystem/Component/Rad.svelte [hunk 4]
```diff
@@ -1,10 +1,10 @@
 <script>
-  import { Icon, Numeric } from "../Primitive";
+  import { Icon, Title } from "../Primitive";
 
   export let rad = null;
   export let usd = null;
   export let style = null;
-  export let size = "regular"; // regular | big
+  export let deposit = null;
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/DesignSystem/Component/Rad.svelte:7]: 👍

File: ui/DesignSystem/Primitive/Icon/Register.svelte [hunk 5]
```diff
@@ -7,32 +7,28 @@
   {style}
   width="24"
   height="24"
-  fill="var(--color-foreground-level-5)"
   viewBox="0 0 24 24"
+  fill="none"
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/DesignSystem/Primitive/Icon/Register.svelte:11]: 👍

File: ui/Screen/OrgRegistration.svelte [hunk 6]
```diff
@@ -94,45 +94,51 @@
...
-          size="small"
-          variant="square"
-          dataCy="avatar" />
+<style>
+  .wrapper {
+    margin: 92px 0 32px 0;
+  }
+</style>
+
+<ModalLayout dataCy="org-reg-modal">
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/Screen/OrgRegistration.svelte:103]: :+1:

File: ui/DesignSystem/Component/Transaction.svelte [hunk 7]
```diff
@@ -25,19 +30,18 @@
...
-      default:
-        return "square";
-    }
-  };
 
   const updateAvatar = async () => (avatar = await subject.avatarSource);
 
+  let copyIcon = Icon.Copy;
+
+  const afterCopy = () => {
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/DesignSystem/Component/Transaction.svelte:38]: was  thinking the same thing

File: ui/DesignSystem/Component/Rad.svelte (hunk 4 above)
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/DesignSystem/Component/Rad.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/637/commits/7eafea9375ad5ad780bf99e08811a60f9ffce9b5

File: ui/DesignSystem/Primitive/Icon/Register.svelte (hunk 5 above)
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/DesignSystem/Primitive/Icon/Register.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/637/commits/566005328a4ca4e3d8c397ae9832f58c882f35b8

File: ui/Screen/OrgRegistration.svelte (hunk 6 above)
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/Screen/OrgRegistration.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/637/commits/fa101008f0c0bb532c271c9a278c4e78bdcc8e0e

File: ui/DesignSystem/Component/Transaction/Header.svelte (hunk 2 above)
Comment (juliendonck) [reply ui/DesignSystem/Component/Transaction/Header.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/637/commits/a79cd89e9bc466e424ad1a72718a7a71b110a01e

File: ui/DesignSystem/Component/Transaction.svelte (hunk 7 above)
[the remark this answers is not in the document]
Comment (rudolfs) [reply ui/DesignSystem/Component/Transaction.svelte]: Doing it was faster than creating an issue: https://github.com/radicle-dev/radicle-upstream/pull/637/commits/0702dad7a730cfa2bc0c1905fe2567efbe2fe767.

Proof:

Screenshot 2020-07-09 at 15 26 08
Screenshot 2020-07-09 at 15 26 15
Screenshot 2020-07-09 at 15 26 27
Screenshot 2020-07-09 at 15 26 43
Screenshot 2020-07-09 at 15 26 49
Screenshot 2020-07-09 at 15 27 08

File: ui/DesignSystem/Primitive/Title.svelte (hunk 3 above)
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/DesignSystem/Primitive/Title.svelte:5]: `h1 > h2 > h3 > h4 >h5 > h6` That's what we use in the design file.

**Pull request merged**